content/cloud/variables.md

```diff
@@ -0,0 +1,191 @@
+
+This tutorial will guide you through the process of creating a simple application that validates passwords.
+
+## Prerequisites
+
+Before starting, ensure that you have Spin installed on your computer. You can follow the official Fermyon Cloud Quickstart guide to [install Spin](/cloud/quickstart#install-spin) and [log into Fermyon Cloud](/cloud/quickstart#log-in-to-the-fermyon-cloud).
+
+Since this example is written in Python, make sure you have the required tools installed for Python Spin applications. The Spin CLI facilitates the creation of new Spin applications using application [templates](/cloud/cli-reference#templates). In this tutorial, we will use the `http-py` template. Install it along with the `py2wasm` Spin plugin, which enables compiling Python apps to Wasm, using the following commands:
+
+<!-- @selectiveCpy -->
+
+```bash
+$ spin templates install --git https://github.com/fermyon/spin-python-sdk --upgrade
+$ spin plugins install py2wasm --yes
+```
+
+The output from the above commands should be similar to the following:
+
+<!-- @nocpy -->
+
+```text
+Copying remote template source
+Installing template http-py...
+Installed 1 template(s)
+
++---------------------------------------------+
+| Name      Description                       |
++=============================================+
+| http-py   HTTP request handler using Python |
++---------------------------------------------+
+
+Plugin 'py2wasm' was installed successfully!
+
+Description:
+	A plugin to convert Python applications to Spin compatible modules
+
+Homepage:
+	https://github.com/fermyon/spin-python-sdk
+```
+
+## Creating Our Spin Application
+
+Let's start by [creating a new Spin application](/cloud/cli-reference#new) using the Python template. Follow the steps below:
+
+<!-- @selectiveCpy -->
+
+```bash
+$ spin new http-py pw_checker
+Project description: A Spin app that validates a password
+HTTP base: /
+HTTP path: /...
+```
+
+## Adding Variables to an Application Component
+
+Our application receives a password via the HTTP request body, compares it to an expected password, and returns a JSON response indicating whether the submitted password matched or not.
+
+In reality, you'd have multiple usernames and password hashes in a database, but for this tutorial we will configure the expected password using a Spin application variable. We'll name the variable `password` and set `required = true` since there is no reasonable default value. To do this, add a top-level `[variables]` section to the application manifest (`spin.toml`) and declare the variable within it.
+
```

Comment:
I would have preferred to have the initial instruction being illustrated here. When I read this, I would go to the sample below, before reading the next paragraph.

Reply:
what do you mean by initial instruction? Are you saying to have the example manifest displayed between paragraphs?

Reply:
(speaking for myself not Mikkel) Yes, it would be good to show the `variables` stanza here, then the `component.config` stanza under that paragraph - I think this would help readers see what specifically they need to change.  And could show the whole manifest at the bottom as it is currently.  E.g.

_(explanation of variables)_

```
# Add this above the [component] section
[variables]
password = {...}
```

_explanation of component config_

```
# Add this at the end of the file
[component.config]
password = "{{ password }}"
```

_(note about explicitly using variables)_

Your `spin.toml` file should now look like this:

_(full spin.toml)_

Reply:
Exactly 👍🏻 Sorry for not being clear.

Reply:
Thanks team! I like this a lot -- updated!